content/documentation/internals/nats-streaming-protocol.md

```diff
@@ -0,0 +1,199 @@
+| [`CloseResp`](#CLOSERESP)         | Server  | Result of the close request
+
+The following sections explain each protocol message.
+
+## <a name="CONNREQ"></a>ConnectRequest
+
+### Message Structure
+
+* `ClientID`: A unique identifier for a client.
+* `HeartbeatInbox`: An inbox to which the streaming server will send heartbeats to for the client to process.
+
+### Description
+
+A connection request is sent when a streaming client connects to the streaming server. The connection request contains a unique identifier representing the client, and an inbox subject the client will listen on for incoming heartbeats.  The identifier **must** be unique; a connection attempt with an identifier currently in use will fail.  The inbox subject is the subject where the client receives incoming heartbeats, and responds by publishing an empty NATS message to the reply subject, indicating it is alive.  The streaming server will return a [ConnectResponse](#CONNRESP) message to the reply subject specified in the NATS request message.
+
+This request is published to a subject comprised of the `<discover-prefix>.cluster-id`, for example, if a NATS streaming server was started with a cluster-id of `mycluster`, and the default prefix was used, the client publishes to `_STAN.discover.mycluster`
+
+## <a name="CONNRESP"></a>ConnectResponse
+
+### Message Structure
+
+*	`pubPrefix`: Prefix to use when publishing to this streaming server cluster   
+*	`subRequests`: Subject used for subscription requests
+*	`unsubRequests`: Subject used for unsubscribe requests
+*	`closeRequests`: Subject for closing the connection to the streaming server
+*	`error`: An error string, which will be empty/omitted upon success
+*	`publicKey`:  Reserved for future use.
+
+### Description
+After a `ConnectRequest` is published, the streaming server responds with this message on the reply subject of the underlying NATS request.  The streaming server requires the client to make requests and publish messages on certain subjects (described above), and when a connection is successful, the client saves the information returned to be used in sending other NATS streaming protocol messages.  In the event the connection was not successful, an error is returned in the `error` field.
+
+## <a name="SUBREQ"></a>SubscriptionRequest
+
+### Message Structure
+
+* `clientID`: Client ID originally provided to the [ConnectRequest](#CONNREQ)
+* `subject`: Formal subject to subscribe to, e.g. foo.bar
+* `qGroup`: Optional queue group
+* `inbox`: Inbox subject to deliver messages on
+* `maxInFlight`:  Maximum inflight messages without an acknowledgement allowed
+* `ackWaitInSecs`: Timeout for receiving an acknowledgement from the client
+* `durableName`: Optional durable name which survives client restarts
+* `startPosition`: Start position type
+* `startSequence`: Optional start sequence number
+* `startTimeDelta`: Optional start time
+
+StartPosition
```

Comment:
It's a little hard to describe that field accurately in a way that makes sense at first glance. I would try something like this this:  

`startPosition`: An enumerated type specifying which kind of offset will be used to deliver historical data to this subscription.